common/common-utils/src/main/java/com/duckduckgo/common/utils/extensions/FragmentExtensions.kt

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright (c) 2024 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.common.utils.extensions
+
+import android.widget.EditText
+import androidx.fragment.app.Fragment
+
+fun Fragment.showKeyboard(editText: EditText) = requireActivity().showKeyboard(editText)
```

Comment:
We should use `activity?.` instead of `requireActivity` for keyboard operations.

The `Activity` can be `null` when the `Fragment` has been detached, and for this case we don’t care if the keyboard shows or not, it’s about to be destroyed.

The user could bring up the keyboard at the wrong time during the lifecycle or it could be shown in an async operation after the `Activity` is destroyed, both of these scenarios would result in a crash.

So I think we should do something like this instead:

```suggestion
fun Fragment.showKeyboard(editText: EditText) {
    activity?.showKeyboard(editText) ?: Timber.d("FragmentExtensions", "Activity is null, cannot show keyboard")
}
```

This way we still get info but it won’t crash the app for edge cases.

Reply:
Personally, I don’t like defensive programming like this. I’d rather fail fast and know this is not working properly, because if this is failing it’s probably a result of someone calling show/hideKeyboard in the wrong part of the Fragment lifecycle.

I’d worry a log statement might be easily missed by someone if they are using this and it fails, is there any way we can send an error to our crash reporting tool?

Reply:
> is there any way we can send an error to our crash reporting tool?

We won't receive those in the crash tool, only in the logs.

> probably a result of someone calling show/hideKeyboard in the wrong part of the Fragment lifecycle

Not necessarily, the user can initiate `showKeyboard`/`hideKeyboard` while the `Fragment` is being destroyed. So unless we can always insure that this doesn’t happen, I won't be able to approve the change.

Also, we don't care if the keyboard hides/shows when `Activity` = `null`, so it’s not worth crashing the app over. Show/hide keyboard is not a particularly important operation.

Reply:
As we talked about on MM, I disagree with this approach. If a developer is calling showKeyboard without an Activity then something is wrong and they should know about it. The logging statement for me could easily go unnoticed. I appreciate this is more of an obvious situation in that the keyboard should show and they probably would notice, but in general I like to fail fast and early, and know about it.

For the sake of getting this fixed and moving on I’ll make the changes 👍